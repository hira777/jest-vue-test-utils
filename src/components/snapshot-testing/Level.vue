<template>
  <div v-show="level !== ''">
    <p v-show="level === 'good'" ref="good">good</p>
    <p v-show="level === 'veryGood'" ref="veryGood">very good</p>
    <p v-show="level === 'excellent'" ref="excellent">excellent</p>
  </div>
</template>

<script>
export default {
  name: 'Level',
  props: {
    level: {
      type: String,
      default: '',
      validator: value =>
        ['', 'good', 'veryGood', 'excellent'].indexOf(value) !== -1
    }
  }
};
</script>

<style></style>
