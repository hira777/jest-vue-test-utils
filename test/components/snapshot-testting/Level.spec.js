import { shallowMount } from '@vue/test-utils';
import Level from '@/components/snapshot-testing/Level.vue';

describe('Level.vue', () => {
  it('good', () => {
    const wrapper = shallowMount(Level, {
      propsData: { level: 'good' }
    });

    expect(wrapper.find({ ref: 'good' }).isVisible()).toBe(true);
    expect(wrapper.find({ ref: 'veryGood' }).isVisible()).toBe(false);
    expect(wrapper.find({ ref: 'excellent' }).isVisible()).toBe(false);
  });

  it('veryGood', () => {
    const wrapper = shallowMount(Level, {
      propsData: { level: 'veryGood' }
    });

    expect(wrapper.find({ ref: 'good' }).isVisible()).toBe(false);
    expect(wrapper.find({ ref: 'veryGood' }).isVisible()).toBe(true);
    expect(wrapper.find({ ref: 'excellent' }).isVisible()).toBe(false);
  });

  it('excellent', () => {
    const wrapper = shallowMount(Level, {
      propsData: { level: 'excellent' }
    });

    expect(wrapper.find({ ref: 'good' }).isVisible()).toBe(false);
    expect(wrapper.find({ ref: 'veryGood' }).isVisible()).toBe(false);
    expect(wrapper.find({ ref: 'excellent' }).isVisible()).toBe(true);
  });

  it('empty string', () => {
    const wrapper = shallowMount(Level);

    expect(wrapper.isVisible()).toBe(false);
  });

  describe('snapshot', () => {
    it('good', () => {
      const wrapper = shallowMount(Level, {
        propsData: { level: 'good' }
      });
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('veryGood', () => {
      const wrapper = shallowMount(Level, {
        propsData: { level: 'veryGood' }
      });

      expect(wrapper.html()).toMatchSnapshot();
    });

    it('excellent', () => {
      const wrapper = shallowMount(Level, {
        propsData: { level: 'excellent' }
      });

      expect(wrapper.html()).toMatchSnapshot();
    });

    it('empty string', () => {
      const wrapper = shallowMount(Level);

      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
