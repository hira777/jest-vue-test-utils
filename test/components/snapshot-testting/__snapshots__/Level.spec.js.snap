// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Level.vue snapshot empty string 1`] = `
<div style="display: none;">
  <p style="display: none;">good</p>
  <p style="display: none;">very good</p>
  <p style="display: none;">excellent</p>
</div>
`;

exports[`Level.vue snapshot excellent 1`] = `
<div>
  <p style="display: none;">good</p>
  <p style="display: none;">very good</p>
  <p>excellent</p>
</div>
`;

exports[`Level.vue snapshot good 1`] = `
<div>
  <p>good</p>
  <p style="display: none;">very good</p>
  <p style="display: none;">excellent</p>
</div>
`;

exports[`Level.vue snapshot veryGood 1`] = `
<div>
  <p style="display: none;">good</p>
  <p>very good</p>
  <p style="display: none;">excellent</p>
</div>
`;
